### PR TITLE
Prune stale read-state and discovered-state records after provider refresh

### LINKED ISSUE #475: ReadStateStore grows unbounded — stale read-state keys are never pruned
## Problem

`ReadStateStore` persists a set of "read" item keys (composite strings used to track which inbox items the user has viewed) in `globalState` under `devdocket.read-state`. Keys are added whenever the user views an item, but **they are never removed**.

When issues are closed, PRs merge, and providers stop returning those items, the read-state keys for those items remain in `globalState` forever:

```typescript
// packages/core/src/storage/readStateStore.ts
async add(key: string): Promise<boolean> {
  // Adds a key — but there is no corresponding prune/cleanup
}
// No prune() method exists
```

This is the exact same unbounded growth pattern that was fixed for `DiscoveredStateStore` in issue #401, which added a `prune()` method to clean up stale records after provider refreshes. `ReadStateStore` has no equivalent.

For a heavy user over months/years (e.g., hundreds of GitHub issues and ADO work items viewed), the `devdocket.read-state` array in `globalState` accumulates all previously-seen item keys indefinitely.

## Proposed fix

Add a `prune(activeKeys: Set<string>): Promise<number>` method to `ReadStateStore`, mirroring `DiscoveredStateStore.prune()`:

```typescript
async prune(activeKeys: Set<string>): Promise<number> {
  if (!this.loaded) { await this.load(); }
  const staleKeys: string[] = [];
  for (const key of this.items) {
    if (!activeKeys.has(key)) { staleKeys.push(key); }
  }
  if (staleKeys.length === 0) { return 0; }
  for (const key of staleKeys) { this.items.delete(key); }
  await this.persist();
  this._onDidChange.fire();
  return staleKeys.length;
}
```

Call it after provider refreshes, passing the set of currently-active item keys. The key format used by `ReadStateStore` is the same composite key used for inbox items — `getDiscoveredItemKey(providerId, externalId)` — so the active-keys set can be built from the same discovered items map used by `DiscoveredStateStore.prune()`.

The `prune()` call for `ReadStateStore` should be guarded the same way as `DiscoveredStateStore.prune()` — skip pruning when a provider returns zero items to avoid incorrectly pruning during transient API failures.

### PR DESCRIPTION
**Pull request by mthalman**
## Summary

This fixes stale read-state growth by pruning provider-scoped read-state keys after provider refreshes, using the same active discovered-items map as discovered-state cleanup.

## Details

- Add `ReadStateStore.prune(activeItems)` with the same safety semantics as `DiscoveredStateStore.prune`: lazy load, skip providers that returned empty arrays, prune only records for providers with non-empty refresh results, return the removed count, and fire `onDidChange` only when records are removed.
- Wire both `DiscoveredStateStore.prune` and `ReadStateStore.prune` from the production `ProviderRegistry.onDidRefreshProvider` path.
- Reconnect `DiscoveredStateStore.prune`, which was orphaned when the old `InboxTreeProvider` refresh path was retired.
- Cover read-state pruning behavior with unit tests and verify the extension-level refresh wiring for non-empty and empty provider refreshes.

## Validation

- `cd packages\core ; npm run build`
- `npm run test`

Closes #475

### CODE REVIEW

File: packages/core/src/extension.ts [hunk 1]
```diff
@@ -446,6 +446,18 @@ export async function activate(context: vscode.ExtensionContext): Promise<DevDoc
     ws.onDidChangeWatchedRuns(safeHandler('mc:watchedRuns', () => mainProvider.scheduleRefresh())),
     ws.onDidChangePRWatches(safeHandler('mc:watchedPRs', () => mainProvider.scheduleRefresh())),
     readStateStore.onDidChange(safeHandler('mc:readState', () => mainProvider.scheduleRefresh())),
+    pr.onDidRefreshProvider(safeHandler('mc:prune', async () => {
+      const active = pr.getAllDiscoveredItems();
+      try {
+        const ssPruned = await ss.prune(active);
+        const rsPruned = await readStateStore.prune(active);
+        if (ssPruned > 0 || rsPruned > 0) {
+          logger.debug(`Pruned ${ssPruned} discovered-state and ${rsPruned} read-state records`);
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/core/src/extension.ts]: Fixed in b1d19f1
